ext/spl/spl_directory.c

```diff
@@ -1828,34 +1828,6 @@ zend_object_iterator *spl_filesystem_tree_get_iterator(zend_class_entry *ce, zva
 }
 /* }}} */
 
-/* {{{ spl_filesystem_object_cast */
-static int spl_filesystem_object_cast(zend_object *readobj, zval *writeobj, int type)
-{
-	spl_filesystem_object *intern = spl_filesystem_from_obj(readobj);
-
-	if (type == IS_STRING) {
-		if (readobj->ce->__tostring) {
-			return zend_std_cast_object_tostring(readobj, writeobj, type);
-		}
-
-		switch (intern->type) {
-		case SPL_FS_INFO:
-		case SPL_FS_FILE:
-			ZVAL_STRINGL(writeobj, intern->file_name, intern->file_name_len);
-			return SUCCESS;
-		case SPL_FS_DIR:
-			ZVAL_STRING(writeobj, intern->u.dir.entry.d_name);
-			return SUCCESS;
```

Comment:
Can this case really be dropped?

Reply:
The stub file has an implementation alias for the ``__toString`` method on that class

Reply:
Ah, right.